internal/pkg/pluginmanager/downloader_test.go

```diff
@@ -23,5 +23,5 @@ func TestDownload(t *testing.T) {
 func TestDownloadNotFound(t *testing.T) {
 	c := NewDownloadClient()
 	err := c.download(".", "doesntexist", "0.0.1")
-	assert.Contains(t, err.Error(), "404")
+	assert.Contains(t, err.Error(), "403")
```

Comment:
Please add a comment here to explain why "not found" is 403 instead of 404 for an S3 bucket.

Reply:
For public users instead of the group we specified, it is not allowed to have the permission of list bucket, so for nonexistent object, 403 is returned instead of 404.

Reply:
I think it's better to put what you said in the code as a line of comment. Because by default, developers will consider "NOT FOUND" as 404, not 403.

Reply:
Has been add comment in the code comments.